### PR TITLE
Add operator== & operator!= to message generation.

### PR DESCRIPTION
**Pull request by DangitBen**
Solves #33

### CODE REVIEW

File: scripts/msg.h.template [hunk 1]
```diff
@@ -173,6 +173,20 @@ ros::message_operations::Printer< @(cpp_full_name_with_alloc) >::stream(s, "", v
 return s;
 }
 
+@# Equality
+template<typename ContainerAllocator>
+bool operator==(const @(cpp_full_name_with_alloc) & lhs, const @(cpp_full_name_with_alloc) & rhs)
```
Comment (stertingen) [inline scripts/msg.h.template]: Maybe allow comparison between objects using different allocator types?

Something like
```c++
template<typename ContainerAllocator1, typename ContainerAllocator2>
bool operator==(const MsgType<ContainerAllocator1>& lhs, const MsgType<ContainerAllocator2>& rhs)
{}
```

Comment (dirk-thomas) [reply scripts/msg.h.template]: @DangitBen can you please respond to this question.

Comment (DangitBen) [reply scripts/msg.h.template]: Sorry! I guess I lost the notification for @stertingen's question. I think I can add this tomorrow.

Out of curiosity, why is this something that you wanted?

**Comment by dirk-thomas** [comment]
I just improved the readability of the generated code a bit: b1528ffd5bca181faf044432d477085dafda4fbe.

Thank you for the patch.

**Pull request merged**